napari/layers/image/image.py

```diff
@@ -692,6 +698,43 @@ def _set_view_slice(self):
             self.reset_contrast_limits()
             self._should_calc_clims = False
 
+    def _project_slice(self):
+        """
+        Project a thick nD slice into a single 2d or 3d image that can be passed to the GPU.
+
+        Projection is based on slice thickness and the current projection mode.
+        """
+        if self.slice_projection == ImageSliceProjection.NONE:
+            return self.data[self._slice_indices]
+
+        half_thickness = (
+            np.maximum(np.ceil(self._thickness).astype(int), 1) / 2
+        )
+        image_slices = tuple(
+            slice(
+                ceil(max(0, self._slice_indices[i] - half_thickness[i])),
+                ceil(
+                    min(
+                        self._slice_indices[i] + half_thickness[i],
+                        self.data.shape[i],
+                    )
+                ),
+            )
+            if i in self._dims_not_displayed
+            else self._slice_indices[i]
+            for i in range(self.ndim)
+        )
+
+        proj_funcs = {
+            ImageSliceProjection.ADDITIVE: np.sum,
+            ImageSliceProjection.AVERAGE: np.mean,
+        }
```

Comment:
This should be created/ stored outside of the function

Reply:
Yeah, I wasn't sure where to put it. Should this live somewhere in image utils maybe?

Reply:
image utils sounds good